RecoMET/METProducers/src/CaloRecHitsBeamHaloCleaned.cc

```diff
@@ -0,0 +1,211 @@
+// -*- C++ -*-
+//
+// Package:    RecoMET/METProducers
+// Class:      CaloRecHitsBeamHaloCleaned
+// 
+/**\class CaloRecHitsBeamHaloCleaned CaloRecHitsBeamHaloCleaned.cc RecoMET/METProducers/plugins/CaloRecHitsBeamHaloCleaned.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Thomas Laurent
+//         Created:  Tue, 09 Feb 2016 13:09:37 GMT
+//
+//
+
+
+// system include files
+#include <memory>
+
+// user include files
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/stream/EDProducer.h"
+
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/Utilities/interface/StreamID.h"
+#include <vector>
+#include <iostream>
+#include "DataFormats/Math/interface/Point3D.h"
+#include "DataFormats/TrackReco/interface/Track.h"
+#include "DataFormats/TrackReco/interface/TrackFwd.h"
+#include "DataFormats/METReco/interface/GlobalHaloData.h"
+#include "RecoMET/METAlgorithms/interface/GlobalHaloAlgo.h"
+#include "DataFormats/EcalDetId/interface/EBDetId.h"
+#include "DataFormats/EcalDetId/interface/EEDetId.h"
+#include "Geometry/CaloGeometry/interface/CaloGeometry.h"
+#include "Geometry/Records/interface/CaloGeometryRecord.h"
+
+using namespace reco;
+
+class CaloRecHitsBeamHaloCleaned : public edm::stream::EDProducer<> {
```

Comment:
@slava77 It's not used so far but the plan is to use this producer in a trigger path. See this JIRA ticket: 
https://its.cern.ch/jira/browse/CMSHLT-737
I first need to have it integrated in CMSSW before I can submit the path.